docs/Releases.md

```diff
@@ -13,6 +13,307 @@ use a date based versioning system.  Now, a release version can look
+
+The NF accepts a json config with these rules:
+```json
+"ruleName": {
+		"ip": "127.1.1.0",
+		"depth": 32,
+		"action": 0
+}
+```
+
+
+### Payload Search NF:
+The Payload Scan NF provides the functionality to search for a string within a given UDP or TCP packet payload. Packet is forwarded to its destination NF on a match, dropped otherwise.
+
+### TTL Flags:
+Adds TTL and packet limit flags to stop the NF or the onvm_mgr based on time since startup or based on packets received. Default measurements for these flags are in seconds and in millions of packets received. 
+
+Usage:  
+ - `-t TTL` will stop the NF after the NF tx exceeds `PKT_LIMIT * PKT_TTL_MULTIPLIER`, measured in seconds. 
+ - `-l PACKET_LIMIT` will stop the NF after the NF was alive longer than TTL_TIME*TIME_TTL_MULTIPLIER, measured in millions of packets.
+
+### NF to NF Messaging:
+Adds the ability for NFs to send messages to other NFs. NFs need to define a message handler to receive messages and are responsible to
+free the custom message data. If the message is sent to a NF that doesn't have a message handler the message is ignored.
+
+### Minor improvements
+ - **Make Number of mbufs a Constant Value** - Previously the number of mbufs was calculated based on the `MAX_NFS` constant. This lead to performance degradation as the requested number of mbufs was too high, changing this to a constant has significantly improved performance.  
+ - **Reuse NF Instance IDs** - Reuse instance IDs of old NFs that have terminated. The instance IDs are still continiously incremented up to the `MAX_NFS` constant, but when that number is reached the next NF instance ID will be wrapped back to the starting value and find the first unoccupied instance ID.    
+ - Check if ONVM_HOME is Set Before Compiling ONVM   
+ - Add Core Information to Web Stats
+ - Update Install Script Hugepage Setup & Kernel Driver Installation  
+ - Add Compatibility Changes to Run ONVM on Ubuntu 18.04.1  
+ - Various Documentation updates and fixes  
+ - Change onvm-pktgen Submodule to Upstream Pktgen  
+
+### Bug fixes:
+ - Free Memory on ONVM_MGR Shutdown  
+ - Launch Script to Handle Multi-word String Arguments  
+ - NF Advanced Ring Thread Process NF Shutdown Messages  
+ - Adds NF Ring Cleanup Logic On Shutdown  
+ - Resolve Shutdown Memory Leaks  
+ - Add NF Tag Memory Allocation  
+ - Fix the Parse IP Helper Function  
+ - Fix Speed Tester NF Generated Packets Counter  
+ - Add Termination of Started but not yet Running NFs  
+ - Add ONVM mgr web mode memory cleanup on shutdown  
+ - Removes the Old Flow Tracker NF Launch Script  
+ - Fix Deprecated DPDK Function in Speed Tester NF  
+
+**v19.05 API Changes:**
```

Comment:
You should move the API changes part in the arch changes portion that I commented on above to here